.github/workflows/CI.yml

```diff
@@ -360,40 +434,55 @@ jobs:
           cp .aoc-stack-test.yml build/packages/
           cp .aoc-stack-release.yml build/packages/
 
-      - name: Cache the packages
-        uses: actions/cache@v2
-        with:
-          key: "cached_packages_${{ github.run_id }}"
-          path: build/packages
-
   e2etest-release:
     runs-on: ubuntu-latest
     needs: [e2etest-preparation]
 
     steps:
       - uses: actions/checkout@v2
 
+      - name: Cache if success
+        id: e2etest-release
+        uses: actions/cache@v2
+        with:
+          path: |
+            VERSION
+          key: e2etest-release-${{ github.run_id }}
+
       - name: Configure AWS Credentials
+        if: steps.e2etest-release.outputs.cache-hit != 'true'
         uses: aws-actions/configure-aws-credentials@v1
         with:
           aws-access-key-id: ${{ secrets.INTEG_TEST_AWS_KEY_ID }}
           aws-secret-access-key: ${{ secrets.INTEG_TEST_AWS_KEY_SECRET }}
           aws-region: us-west-2
 
       - name: restore cached rpms
+        if: steps.e2etest-release.outputs.cache-hit != 'true'
         uses: actions/cache@v2
         with:
           path: build/packages
           key: "cached_packages_${{ github.run_id }}"
 
+      - name: Create SSM package
+        if: steps.e2etest-release.outputs.cache-hit != 'true'
+        run: |
+          ssm_pkg_version=`cat build/packages/VERSION`
+          ssm_pkg_version="${ssm_pkg_version:1}"
+          python3 tools/ssm/ssm_manifest.py ${ssm_pkg_version}
+          aws s3 cp build/packages/ssm s3://aws-otel-collector-test/ssm --recursive
+          bash tools/ssm/ssm_create.sh testAWSDistroOTel-Collector ${ssm_pkg_version} aws-otel-collector-test/ssm
```

Comment:
We might need to clean up the document though, the default quota for ssm accounts is not very high. Or we can remove the test ssm document if all the ec2 test has passed.

Reply:
Agree. I need to verify what happen when reaching SSM quota. We could add the clean up logic in  resource-clean.yml later.

Reply:
The SSM quota issue will be addressed in a separate PR.